### PR TITLE
fix include

### LINKED ISSUE #1585: c_api.h uses undefined THREAD_LOCAL macro
## Environment info

Operating System: x86_64-linux-gnu

CPU: Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz

C++/Python/R version: gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.9)

## Error message

/usr/local/include/LightGBM/c_api.h: In function ‘char* LastErrorMsg()’:
/usr/local/include/LightGBM/c_api.h:765:38: error: ‘THREAD_LOCAL’ does not name a type
 static char* LastErrorMsg() { static THREAD_LOCAL char err_msg[512] = "Everything is fine"; return err_msg; }
                                      ^
/usr/local/include/LightGBM/c_api.h:765:100: error: ‘err_msg’ was not declared in this scope
 static char* LastErrorMsg() { static THREAD_LOCAL char err_msg[512] = "Everything is fine"; return err_msg; }

## Reproducible examples

In any source file, including <LightGBM/c_api.h>
results in compile error about undefined THREAD_LOCAL

The only workaround is to include <LightGBM/utils/log.h> before including <LightGBM/c_api.h>. Moving the definition of THREAD_LOCAL to c_api.h (or export.h) would fix the issue.

### PR DESCRIPTION
**Pull request by guolinke**
fix #1585

### CODE REVIEW

File: include/LightGBM/c_api.h [hunk 1]
```diff
@@ -2,6 +2,7 @@
 #define LIGHTGBM_C_API_H_
 
 #include <LightGBM/meta.h>
```
Comment (alokpr) [inline include/LightGBM/c_api.h]: Is this file supposed to expose a c-api, which can be used by a C program?
If so, including header files with C++ code will not compile. I believe both - meta.h and log.h - have C++ code?
Other C++ stdlib headers (cstdint, exception, etc) included in this file would also not compile.

Comment (guolinke) [reply include/LightGBM/c_api.h]: We don't think it is used for c program.

Comment (alokpr) [reply include/LightGBM/c_api.h]: Forgive my ignorance. Why is it called "C" API then, when it cannot be used from a "C" program?

Comment (guolinke) [reply include/LightGBM/c_api.h]: that is for python package abd other language to call. Anyway, I just reduce the dependence of head files.

**Pull request merged**